src/monitor-plan/monitor-plan.service.ts

```diff
@@ -35,6 +35,17 @@ export class MonitorPlanService {
     results.sort((a, b) => {
       return (a.name < b.name) ? -1 : (a.name == b.name) ? 0 : 1
     });
-    return results;
-  }
+    return this.setMonitoringPlanStatus(results);
+    }
+
+
+ setMonitoringPlanStatus(MonitoringPlanConfiguration: MonitorPlanDTO[]): MonitorPlanDTO[] {
+    MonitoringPlanConfiguration.forEach(mp => {
+      if(mp.endReportPeriodId == null){
+        mp.active = true;
+        delete mp['endReportPeriodId'];
```

Comment:
Look at using the @Exclude() decorator on the DTO property to exclude it from being serialized back to the client

@Exclude()
endReportPeriodId: number;